src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -20,9 +20,11 @@ public static class ValidationUtilities
 
         private static readonly Regex TrackedResRegEx = new Regex(@".+/Resource$", RegexOptions.IgnoreCase);
 
-        // This needs to be deprecated in favor of context.TrackedResources
+        public static readonly Regex ResourcePathPattern = new Regex(@"/providers/(?<providerNamespace>[^{/]+)((/(?<resource>[^{/]+)/)((?<resourceName>[^/]+)))+(/(?<resource>[^{/]+))");
+
         private static readonly Regex resourceProviderPathPattern = new Regex(@"/providers/(?<resPath>[^{/]+)/", RegexOptions.IgnoreCase);
 
+        // This needs to be deprecated in favor of context.TrackedResources
         public static bool IsTrackedResource(Schema schema, Dictionary<string, Schema> definitions)
```

Comment:
This method is being referenced by existing tracked resources validation rules that @sarangan12 is working on improving, I have instructed him to remove this code once he has his changes ready